lib/src/main/java/org/team100/lib/commands/drivetrain/DriveWithProfile2.java

```diff
@@ -112,22 +113,23 @@ public void execute() {
         Rotation2d currentRotation = m_swerve.getState().pose().getRotation();
         // take the short path
         double measurement = currentRotation.getRadians();
-        Optional<Pose2d> opt = m_fieldRelativeGoal.get();
+        Optional<Pose2d> opt = m_fieldRelativeGoalSupplier.get();
         if (opt.isEmpty()) {
+            if (m_fieldRelativeGoal == null) 
```

Comment:
this logic seems to make the command end one cycle after the target is lost.

Reply:
This part is mainly to prevent the case where when I press the button no note is identified having some random result. This will only happen if no object has ever been detected since the robot has been booted due to the fact it only is null upon initialization of the class and is changed upon the first initialization of the command given a pose is given.

Reply:
hm, I think it's not doing what you think it's doing.  maybe start with something simple that can work perfectly.